From 2ef3299c69cd2c240f7f5fb963ced8187ffa51b2 Mon Sep 17 00:00:00 2001
From: Maksym Sditanov <maxim.sditanov@lge.com>
Date: Mon, 6 Jan 2014 12:43:12 +0000
Subject: [PATCH 1/1] Add pbnjson dependency into CMakeLists.txt

:Release Notes:
PmLogLib require to link with pbnjson library.

:Detailed Notes:

:Testing Performed:

:QA Notes:

:Issues Addressed:

Open-webOS-DCO-1.0-Signed-off-by: Maksym Sditanov <maxim.sditanov@lge.com>

Change-Id: Ief5fe8586579cdc6e3ffd2b6de89356e62bca8f6
---
 CMakeLists.txt                  | 5 +++++
 src/ls-hubd/CMakeLists.txt      | 2 +-
 src/ls-hubd/test/CMakeLists.txt | 2 +-
 3 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 038faf2..043b0cf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,6 +43,11 @@ pkg_check_modules(CJSON REQUIRED cjson)
 include_directories(${CJSON_INCLUDE_DIRS})
 webos_add_compiler_flags(ALL ${CJSON_CFLAGS_OTHER})
 
+# check if we have pbnjson
+pkg_check_modules(PBNJSON REQUIRED pbnjson_c)
+include_directories(${PBNJSON_INCLUDE_DIRS})
+webos_add_compiler_flags(ALL ${PBNJSON_CFLAGS_OTHER})
+
 # check if we have PmLogLib
 pkg_check_modules(PMLOGLIB REQUIRED PmLogLib)
 include_directories(${PMLOGLIB_INCLUDE_DIRS})
diff --git a/src/ls-hubd/CMakeLists.txt b/src/ls-hubd/CMakeLists.txt
index 989558d..c8cee9c 100644
--- a/src/ls-hubd/CMakeLists.txt
+++ b/src/ls-hubd/CMakeLists.txt
@@ -37,7 +37,7 @@ add_definitions(-DDEFAULT_LOCAL_SOCKET_DIR="${WEBOS_INSTALL_RUNTIMEINFODIR}/ls2"
 add_library(ls-hublib STATIC ${HUB_SOURCE_FILES})
 
 add_executable(ls-hubd hub.c)
-target_link_libraries(ls-hubd ls-hublib ${CMAKE_PROJECT_NAME} ${PMLOGLIB_LDFLAGS})
+target_link_libraries(ls-hubd ls-hublib ${CMAKE_PROJECT_NAME} ${PMLOGLIB_LDFLAGS} ${PBNJSON_LDFLAGS})
 webos_build_daemon(NAME ls-hubd LAUNCH ${CMAKE_SOURCE_DIR}/files/launch/ ${LS2_RESTRICTED})
 
 add_subdirectory(test)
diff --git a/src/ls-hubd/test/CMakeLists.txt b/src/ls-hubd/test/CMakeLists.txt
index 5961713..3394ee3 100644
--- a/src/ls-hubd/test/CMakeLists.txt
+++ b/src/ls-hubd/test/CMakeLists.txt
@@ -31,6 +31,6 @@ add_definitions(-DUNIT_TESTS)
 
 foreach (TEST ${UNIT_TEST_SOURCES})
     add_executable(${TEST} ${TEST}.c ../hub.c)
-    target_link_libraries(${TEST} ls-hublib ${CMAKE_PROJECT_NAME} ${TESTLIBNAME} ${PMLOGLIB_LDFLAGS})
+    target_link_libraries(${TEST} ls-hublib ${CMAKE_PROJECT_NAME} ${TESTLIBNAME} ${PMLOGLIB_LDFLAGS} ${PBNJSON_LDFLAGS})
     add_test(${TEST} ${TEST})
 endforeach ()
-- 
1.8.3.2

